sqrl-tools/sqrl-config/src/main/java/com/datasqrl/config/SqrlConfigCommons.java

```diff
@@ -344,28 +351,70 @@ public static SqrlConfig fromFiles(ErrorCollector errors, @NonNull Path firstFil
     ErrorCollector localErrors = errors.withConfig(firstFile);
     try {
       Configuration baseconfig = configs.fileBased(JSONConfiguration.class, firstFile.toFile());
+      if (!isValid(baseconfig)) throw new RuntimeException(firstFile + " has validation errors.");
 
       if (otherFiles.length>0) {
         NodeCombiner combiner = new OverrideCombiner();
         CombinedConfiguration cc = new CombinedConfiguration(combiner);
         for (int i = otherFiles.length-1; i >= 0; i--) { //iterate backwards so last file takes precedence
-          localErrors = errors.withConfig(otherFiles[i]);
-          Configuration nextconfig = configs.fileBased(JSONConfiguration.class, otherFiles[i].toFile());
-          cc.addConfiguration(nextconfig);
+          Path otherFile = otherFiles[i];
+          localErrors = errors.withConfig(otherFile);
+          Configuration nextConfig = configs.fileBased(JSONConfiguration.class, otherFile.toFile());
+          if (!isValid(nextConfig)) throw new RuntimeException(otherFile + " has validation errors.");
+          cc.addConfiguration(nextConfig);
         }
         cc.addConfiguration(baseconfig);
         resultconfig = cc;
       } else {
         resultconfig = baseconfig;
       }
+
       String configFilename = firstFile.toString();
-      return
-          new SqrlConfigCommons(errors.withConfig(configFilename), configFilename, resultconfig, "");
+      return new SqrlConfigCommons(errors.withConfig(configFilename), configFilename, resultconfig, "");
     } catch (ConfigurationException e) {
       throw localErrors.handle(e);
     }
   }
 
+  @SneakyThrows
+  private static boolean isValid(Configuration baseconfig) {
```

Comment:
SqrlConfigCommons is a general configuration class that we use to read all kinds of configuration files. The logic for validating that a file has a particular schema needs to live in a separate class to not mix concerns.